tracks/javascript/exercises/triangle/mentoring.md

```diff
@@ -0,0 +1,84 @@
+This is the one of the four side exercises unlocked after finishing Resistor Color. The level of understanding is vary from students. Some work on this exercise right after Resistor Color is done, others submit this exercise after solving multiple core exercise.
+
+so you need to teach at least:
+
+- [Logical Operators][ref-logical]
+
+but you might also want to teach according to students' understanding but they'll be taught in future core exercise so you don't need to stick to them.
+
+- [Rest parameters][ref-rest-parameters]
+- [Destructuring assignment][refs-destructuring-assignment]
+- [Array#every][ref-array-every]
+- [Array#some][ref-array-some]
+
+
+Also, this exercise might be the first time for some students to use class, so you gently teach them about JavaScript class if needed.
+
+### Reasonable Solutions
+
+```javascript
+export class Triangle {
+  constructor(...sides) {
+    this.sides = sides;
+  }
+
+  isEquilateral() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && a == b && b == c && c == a;
+  }
+
+  isIsosceles() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && (a == b || b == c || c == a);
+  }
+
+  isScalene() {
+    const [a, b, c] = this.sides;
+    return this.isValid() && a != b && b != c && c != a;
```

Comment:
should we make these strict comparisons?

Reply:
I'd say yes.